File: src/main/java/org/jukeboxmc/inventory/HopperInventory.java
package org.jukeboxmc.inventory;

import com.nukkitx.protocol.bedrock.data.inventory.ContainerType;
import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.blockentity.BlockEntityHopper;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class HopperInventory extends ContainerInventory {

    public HopperInventory( InventoryHolder holder) {
        super( holder, -1, 5 );
    }

    @Override
    public BlockEntityHopper getInventoryHolder() {
        return (BlockEntityHopper) this.holder;
    }

    @Override
    public @NotNull InventoryType getType() {
        return InventoryType.HOPPER;
    }

    @Override
    public @NotNull ContainerType getWindowTypeId() {
        return ContainerType.HOPPER;
    }
}

File: src/main/java/org/jukeboxmc/inventory/HopperInventory.kt
package org.jukeboxmc.inventory

import com.nukkitx.protocol.bedrock.data.inventory.ContainerType
import org.jukeboxmc.blockentity.BlockEntityHopper

/**
 * @author LucGamesYT
 * @version 1.0
 */
class HopperInventory(holder: InventoryHolder?) : ContainerInventory(holder, -1, 5) {
    override val inventoryHolder: InventoryHolder
        get() = holder as BlockEntityHopper
    override val type: InventoryType
        get() = InventoryType.HOPPER
    override val windowTypeId: ContainerType
        get() = ContainerType.HOPPER
}
